packages/@react-spectrum/s2/stories/ActionButton.stories.tsx

```diff
@@ -198,3 +201,41 @@ export const Avatars: Story = {
     );
   }
 };
+
+const NotificationBadgesExample = (args) => {
+  let badgeValue = 10;
+  let formattedValue = useNumberFormatter().format(badgeValue);
+
+  return (
+    <div style={{display: 'flex', gap: 8, padding: 8, justifyContent: 'center'}}>
+      <ActionButton aria-label="Messages has new activity" {...args}><CommentIcon /><NotificationBadge /></ActionButton>
+      <ActionButton aria-label={`${formattedValue} notifications`} {...args}><BellIcon /><NotificationBadge value={badgeValue} /></ActionButton>
+      <ActionButton {...args}><CommentIcon /><Text>Messages</Text><NotificationBadge value={5} /></ActionButton>
+      <ActionButton {...args}><Text>Notifications</Text><NotificationBadge value={105} /></ActionButton>
+    </div>
+  );
+};
+
+export const NotificationBadges: Story = {
+  render: NotificationBadgesExample
+};
+
+NotificationBadges.parameters = {
+  docs: {
+    source: {
+      transform: () => {
+        return `let badgeValue = 10;
+        let formattedValue = useNumberFormatter().format(badgeValue);
+      
+        return (
+          <div style={{display: 'flex', gap: 8, padding: 8, justifyContent: 'center'}}>
+            <ActionButton aria-label="Messages has new activity" {...props}><CommentIcon /><NotificationBadge /></ActionButton>
+            <ActionButton aria-label={\`\${formattedValue} notifications\`}  {...props}><BellIcon /><NotificationBadge value={badgeValue} /></ActionButton>
+            <ActionButton {...props}><CommentIcon /><Text>Messages</Text><NotificationBadge value={5} /></ActionButton>
+            <ActionButton {...props}><Text>Notifications</Text><NotificationBadge value={105} /></ActionButton>
+          </div>
+        )`;
```

Comment:
you might need to new line the `let`? Getting this
<img width="1105" alt="image" src="https://github.com/user-attachments/assets/ac3bad08-2767-4aa0-a224-2c0e929dfb9f" />